### PR TITLE
feat(sidenav component): update Prototype SideNav Component for Produ…

### PR DESCRIPTION
**Pull request by kramduckner**
…ction Usage

- Modify prototype SideNav component to use a nav item list array passed as a prop
- Add unit tests for SideNav component
- Update Styling of SideNav component to utilize updated designs 
- Migrate SideNav component files from /prototype to /navigation

Issue #698

# Pull Request Template

## Issue Overview

This PR addresses [issue #698](https://github.com/bloom-housing/bloom/issues/698)

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds a side nav component https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=6020%3A183885

## How Can This Be Tested/Reviewed?

1. open storybook
2. compare to https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=6020%3A183885

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by jaredcwhite** [comment]
Looks cool @kramduckner! I'm not sure what the timetable is for this, but I'd love to see it enhanced with the "2nd gen" component style/docs. See https://github.com/bloom-housing/bloom/blob/dev/docs/Styling2ndGen.md

**Comment by kramduckner** [comment]
@jaredcwhite  I'd love to those 2nd gen items in there! Could be a great way for me to learn it. I'm not sure who to talk to or what the timeline is to decide if that's something that should be done in this commit or for another ticket/time?

**Comment by emilyjablonski** [comment]
@jaredcwhite I'd love to handle that in a follow-up, make even immediately following this, just to reduce scope of Mark's first PR :)

File: ui-components/src/navigation/SideNav.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const SideNav = (props: SideNavProps) => {
+  const { LinkComponent } = React.useContext(NavigationContext)
+
+  return (
+    <nav className="side-nav" aria-label="Secondary navigation">
+      <ul>
+        {props.navItems?.map((navItem: SideNavItemProps, index: number) => {
+          if (navItem.current) {
+            return (
+              <li className="is-current" key={index}>
```
Comment (emilyjablonski) [inline ui-components/src/navigation/SideNav.tsx:24]: For a11y, we should add `aria-selected` the current element

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Screen Shot 2022-06-06 at 2 47 20 PM
Very close! The tests, stories, and prop set looks great. Looking at the Figma designs (I don't think I had seen those earlier - my b!) it looks like there are a few styling changes to make: (1) The `current` styled state should not have a blue background, and should have a left blue border and (2) there should not be an extra border on the bottom of the nav.

File: ui-components/src/navigation/SideNav.tsx (hunk 1 above)
Comment (kramduckner) [reply ui-components/src/navigation/SideNav.tsx:24]: Great catches! I'll get those styles updated and add that attribute. Thanks @emilyjablonski

**Review by emilyjablonski — APPROVED** [review]
LGTM!! 🥳 🥳 🥳

**Pull request merged**